battlepass/src/benchmarking.rs

```diff
@@ -0,0 +1,171 @@
+#![cfg(feature = "runtime-benchmarks")]
+
+use super::*;
+use crate::Pallet as BPass;
+
+use frame_benchmarking::{benchmarks, impl_benchmark_test_suite, whitelisted_caller};
+use frame_system::RawOrigin;
+use sp_runtime::{DispatchError, traits::SaturatedConversion};
+use sp_std::vec;
+
+const DEPOSIT_AMOUNT: u128 = 10_000_000_000_000_000_000;
+
+/// Fund account with tokens, needed for org and campaign interactions
+fn fund_account<T: Config>(account_id: &T::AccountId) -> Result<(), DispatchError> {
+    let balance_amount: T::Balance = DEPOSIT_AMOUNT.saturated_into();
+	<T as pallet::Config>::Currency::deposit(T::NativeTokenId::get(), &account_id, balance_amount)?;
+	<T as pallet::Config>::Currency::deposit(T::ProtocolTokenId::get(), &account_id, balance_amount)?;
+	Ok(())
+}
+
+/// Fund account with tokens, needed for org and battlepass interactions
+fn get_funded_caller<T: Config>() -> Result<T::AccountId, DispatchError> {
+    let caller: T::AccountId = whitelisted_caller();
+    fund_account::<T>(&caller)?;
+
+    Ok(caller)
+}
+
+
+fn get_org<T: Config>(caller: T::AccountId) -> T::Hash {
+    let org_id = T::ControlBenchmarkHelper::create_org(caller.clone()).unwrap();
+    let treasury_id = T::Control::org_treasury_account(&org_id).unwrap();
+	fund_account::<T>(&treasury_id).unwrap();
+
+    org_id
+}
+
+fn get_battlepass<T: Config>(caller: T::AccountId, org_id: T::Hash) -> T::Hash {
+    let str = BoundedVec::truncate_from(vec![1,2]);
+    let battlepass = types::Battlepass {
+        creator: caller.clone(),
+        org_id,
+        name: str.clone(),
+        cid: str.clone(),
+        season: 1,
+        price: 10,
+        collection_id: 0
+    };
+
+    let _ = BPass::<T>::create_battlepass(RawOrigin::Signed(caller.clone()).into(), org_id, battlepass.cid.clone(), battlepass.name.clone(), battlepass.price.clone());
+
+    <T as frame_system::Config>::Hashing::hash_of(&battlepass)
+}
+
+fn activate_bpass<T: Config>(caller: T::AccountId, battlepass_id: T::Hash) {
+    let _ = BPass::<T>::activate_battlepass(RawOrigin::Signed(caller).into(), battlepass_id);
```

Comment:
I would suggest to check that extrinsic executed successfully. Here and for other helpers as well.